# 01 importing libraries

In [4]:
# Import libraries
import pandas as pd
import numpy as np
import os

In [9]:
df = pd. read_csv(r'C:\Users\Tim\Desktop\Data Analytics Intro\Python Fundamentals for Data Analysts 4\Instacart Basket Analysis\02 Data\Original Data\orders.csv' , index_col = False)

In [10]:
path = r'C:\Users\Tim\Desktop\Data Analytics Intro\Python Fundamentals for Data Analysts 4\Instacart Basket Analysis'

In [11]:
path

'C:\\Users\\Tim\\Desktop\\Data Analytics Intro\\Python Fundamentals for Data Analysts 4\\Instacart Basket Analysis'

In [12]:
df = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'orders.csv'), index_col = False)

In [13]:
df.head()

order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0   2539329        1    prior             1          2                  8   
1   2398795        1    prior             2          3                  7   
2    473747        1    prior             3          3                 12   
3   2254736        1    prior             4          4                  7   
4    431534        1    prior             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [14]:
df.head(30)

order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0    2539329        1    prior             1          2                  8   
1    2398795        1    prior             2          3                  7   
2     473747        1    prior             3          3                 12   
3    2254736        1    prior             4          4                  7   
4     431534        1    prior             5          4                 15   
5    3367565        1    prior             6          2                  7   
6     550135        1    prior             7          1                  9   
7    3108588        1    prior             8          1                 14   
8    2295261        1    prior             9          1                 16   
9    2550362        1    prior            10          4                  8   
10   1187899        1    train            11          4                  8   
11   2168274        2    prior             1          2                 11   
12   1501582        2    prior             2          5                 10   
13   1901567        2    prior             3          1                 10   
14    738281        2    prior             4          2                 10   
15   1673511        2    prior             5          3                 11   
16   1199898        2    prior             6          2                  9   
17   3194192        2    prior             7          2                 12   
18    788338        2    prior             8          1                 15   
19   1718559        2    prior             9          2                  9   
20   1447487        2    prior            10          1                 11   
21   1402090        2    prior            11          1                 10   
22   3186735        2    prior            12          1                  9   
23   3268552        2    prior            13          4                 11   
24    839880        2    prior            14          3                 10   
25   1492625        2    train            15          1                 11   
26   1374495        3    prior             1          1                 14   
27    444309        3    prior             2          3                 19   
28   3002854        3    prior             3          3                 16   
29   2037211        3    prior             4          2                 18   

    days_since_prior_order  
0                      NaN  
1                     15.0  
2                     21.0  
3                     29.0  
4                     28.0  
5                     19.0  
6                     20.0  
7                     14.0  
8                      0.0  
9                     30.0  
10                    14.0  
11                     NaN  
12                    10.0  
13                     3.0  
14                     8.0  
15                     8.0  
16                    13.0  
17                    14.0  
18                    27.0  
19                     8.0  
20                     6.0  
21                    30.0  
22                    28.0  
23                    30.0  
24                    13.0  
25                    30.0  
26                     NaN  
27                     9.0  
28                    21.0  
29                    20.0

In [15]:
df.tail()

order_id  user_id eval_set  order_number  order_dow  \
3421078   2266710   206209    prior            10          5   
3421079   1854736   206209    prior            11          4   
3421080    626363   206209    prior            12          1   
3421081   2977660   206209    prior            13          1   
3421082    272231   206209    train            14          6   

         order_hour_of_day  days_since_prior_order  
3421078                 18                    29.0  
3421079                 10                    30.0  
3421080                 12                    18.0  
3421081                 12                     7.0  
3421082                 14                    30.0

In [16]:
df.shape

(3421083, 7)

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3421083 entries, 0 to 3421082
Data columns (total 7 columns):
 #   Column                  Dtype  
---  ------                  -----  
 0   order_id                int64  
 1   user_id                 int64  
 2   eval_set                object 
 3   order_number            int64  
 4   order_dow               int64  
 5   order_hour_of_day       int64  
 6   days_since_prior_order  float64
dtypes: float64(1), int64(5), object(1)
memory usage: 182.7+ MB


In [18]:
df.dtypes

order_id                    int64
user_id                     int64
eval_set                   object
order_number                int64
order_dow                   int64
order_hour_of_day           int64
days_since_prior_order    float64
dtype: object

In [19]:
df.columns

Index(['order_id', 'user_id', 'eval_set', 'order_number', 'order_dow',
       'order_hour_of_day', 'days_since_prior_order'],
      dtype='object')

In [20]:
df.describe()

order_id       user_id  order_number     order_dow  \
count  3.421083e+06  3.421083e+06  3.421083e+06  3.421083e+06   
mean   1.710542e+06  1.029782e+05  1.715486e+01  2.776219e+00   
std    9.875817e+05  5.953372e+04  1.773316e+01  2.046829e+00   
min    1.000000e+00  1.000000e+00  1.000000e+00  0.000000e+00   
25%    8.552715e+05  5.139400e+04  5.000000e+00  1.000000e+00   
50%    1.710542e+06  1.026890e+05  1.100000e+01  3.000000e+00   
75%    2.565812e+06  1.543850e+05  2.300000e+01  5.000000e+00   
max    3.421083e+06  2.062090e+05  1.000000e+02  6.000000e+00   

       order_hour_of_day  days_since_prior_order  
count       3.421083e+06            3.214874e+06  
mean        1.345202e+01            1.111484e+01  
std         4.226088e+00            9.206737e+00  
min         0.000000e+00            0.000000e+00  
25%         1.000000e+01            4.000000e+00  
50%         1.300000e+01            7.000000e+00  
75%         1.600000e+01            1.500000e+01  
max         2.300000e+01            3.000000e+01

In [21]:
df = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'orders.csv'), nrows=1000)

In [22]:
vars_list = ['order_id', 'user_id', 'order_number', 'order_dow', 'order_hour_of_day', 'days_since_prior_order']

In [23]:
vars_list

['order_id',
 'user_id',
 'order_number',
 'order_dow',
 'order_hour_of_day',
 'days_since_prior_order']

In [24]:
df = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'orders.csv'), usecols = vars_list)

In [25]:
df.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

# df_prods

In [29]:
path = r'C:\Users\Tim\Desktop\Data Analytics Intro\Python Fundamentals for Data Analysts 4\Instacart Basket Analysis\02 Data\Original Data\products.csv'

In [33]:
path = r'C:\Users\Tim\Desktop\Data Analytics Intro\Python Fundamentals for Data Analysts 4\Instacart Basket Analysis'

In [34]:
path

'C:\\Users\\Tim\\Desktop\\Data Analytics Intro\\Python Fundamentals for Data Analysts 4\\Instacart Basket Analysis'

In [ ]:
df = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'products.csv'), index_col = False)

In [35]:
df.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [42]:
df.head(20)

product_id                                       product_name  aisle_id  \
0            1                         Chocolate Sandwich Cookies        61   
1            2                                   All-Seasons Salt       104   
2            3               Robust Golden Unsweetened Oolong Tea        94   
3            4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4            5                          Green Chile Anytime Sauce         5   
5            6                                       Dry Nose Oil        11   
6            7                     Pure Coconut Water With Orange        98   
7            8                  Cut Russet Potatoes Steam N' Mash       116   
8            9                  Light Strawberry Blueberry Yogurt       120   
9           10     Sparkling Orange Juice & Prickly Pear Beverage       115   
10          11                                  Peach Mango Juice        31   
11          12                         Chocolate Fudge Layer Cake       119   
12          13                                  Saline Nasal Mist        11   
13          14                     Fresh Scent Dishwasher Cleaner        74   
14          15                           Overnight Diapers Size 6        56   
15          16                      Mint Chocolate Flavored Syrup       103   
16          17                                  Rendered Duck Fat        35   
17          18                Pizza for One Suprema  Frozen Pizza        79   
18          19   Gluten Free Quinoa Three Cheese & Mushroom Blend        63   
19          20     Pomegranate Cranberry & Aloe Vera Enrich Drink        98   

    department_id  prices  
0              19     5.8  
1              13     9.3  
2               7     4.5  
3               1    10.5  
4              13     4.3  
5              11     2.6  
6               7     4.4  
7               1     1.1  
8              16     7.0  
9               7     8.4  
10              7     2.8  
11              1     9.4  
12             11     1.1  
13             17     6.5  
14             18    11.2  
15             19     5.2  
16             12    17.1  
17              1    10.6  
18              9    12.0  
19              7     6.0

In [43]:
df.tail(35)

product_id                                       product_name  aisle_id  \
0            1                         Chocolate Sandwich Cookies        61   
1            2                                   All-Seasons Salt       104   
2            3               Robust Golden Unsweetened Oolong Tea        94   
3            4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4            5                          Green Chile Anytime Sauce         5   
5            6                                       Dry Nose Oil        11   
6            7                     Pure Coconut Water With Orange        98   
7            8                  Cut Russet Potatoes Steam N' Mash       116   
8            9                  Light Strawberry Blueberry Yogurt       120   
9           10     Sparkling Orange Juice & Prickly Pear Beverage       115   
10          11                                  Peach Mango Juice        31   
11          12                         Chocolate Fudge Layer Cake       119   
12          13                                  Saline Nasal Mist        11   
13          14                     Fresh Scent Dishwasher Cleaner        74   
14          15                           Overnight Diapers Size 6        56   
15          16                      Mint Chocolate Flavored Syrup       103   
16          17                                  Rendered Duck Fat        35   
17          18                Pizza for One Suprema  Frozen Pizza        79   
18          19   Gluten Free Quinoa Three Cheese & Mushroom Blend        63   
19          20     Pomegranate Cranberry & Aloe Vera Enrich Drink        98   

    department_id  prices  
0              19     5.8  
1              13     9.3  
2               7     4.5  
3               1    10.5  
4              13     4.3  
5              11     2.6  
6               7     4.4  
7               1     1.1  
8              16     7.0  
9               7     8.4  
10              7     2.8  
11              1     9.4  
12             11     1.1  
13             17     6.5  
14             18    11.2  
15             19     5.2  
16             12    17.1  
17              1    10.6  
18              9    12.0  
19              7     6.0

In [44]:
df.columns

Index(['product_id', 'product_name', 'aisle_id', 'department_id', 'prices'], dtype='object')

In [46]:
df.shape

(20, 5)

In [47]:
df.dtypes

product_id         int64
product_name      object
aisle_id           int64
department_id      int64
prices           float64
dtype: object

# Step 11
maximum value of the “aisle_id” column = 120
data type of the “department_id” column = int64